framework/math/quadratures/angular/triangle_quadrature.cc

```diff
@@ -0,0 +1,476 @@
+// SPDX-FileCopyrightText: 2024 The OpenSn Authors <https://open-sn.github.io/opensn/>
+// SPDX-License-Identifier: MIT
+
+#include "framework/math/quadratures/angular/triangle_quadrature.h"
+#include "framework/math/quadratures/gausslegendre_quadrature.h"
+#include "framework/math/quadratures/angular/legendre_poly/legendrepoly.h"
+#include "framework/math/math.h"
+#include "framework/logging/log.h"
+#include "framework/runtime.h"
+#include <algorithm>
+#include <cmath>
+#include <iomanip>
+#include <numeric>
+
+namespace opensn
+{
+
+TriangleQuadrature::TriangleQuadrature(unsigned int method, unsigned int sn, unsigned int moments)
+  : method_(method), sn_(sn), moments_(moments)
+{
+  TriangleInit();
+}
+
+void
+TriangleQuadrature::TriangleInit()
+{
+  log.Log0() << "Method: " << method_ << "\nsn " << sn_;
+
+  if (method_ != 1 and method_ != 2 and method_ != 3 and method_ != 0)
+  {
+    log.Log0Error() << "TriangleQuarature: " << method_ << " is not a valid method.\n"
+                    << "Method must be 0, 1, 2, or 3." << std::endl;
+    Exit(510);
+  }
+  if (moments_ < 0 or moments_ > sn_)
+  {
+    log.Log0Error() << "TriangleQuarature: " << moments_ << " must be > 0 and < " << sn_
+                    << std::endl;
+    Exit(510);
+  }
+
+  // Get the gauss points for the z axis. The number of points for GL is twice
+  // that of the quadrature.
+  Vector3 new_omega;
+  const auto GAUSS_LEG_QUAD = GaussLegendreQuadrature(sn_);
+
+  // Formulate the triangular quadrature
+  VecDbl newZi, newWeights;
```

Comment:
I'm guessing `VecDbl` is a typedef for `std::vector<double>`? If so, please use `std::vector<double>`.